torch/distributions/constraint_registry.py

```diff
@@ -138,9 +138,9 @@ def __call__(self, constraint):
         # Look up by Constraint subclass.
         try:
             factory = self._registry[type(constraint)]
-        except KeyError:
+        except KeyError as e:
             raise NotImplementedError(
-                'Cannot transform {} constraints'.format(type(constraint).__name__))
+                'Cannot transform {} constraints'.format(type(constraint).__name__)) from e
```

Comment:
Chaining `KeyError` here is not very meaningful. Also, please use f-string for formatting.
```suggestion
                f'Cannot transform {type(constraint).__name__} constraints') from None
```

Reply:
Sure.

Reply:
@malfet There are a lot of places where `.format(..)` is used instead of f-string. Should I fix all the places where f-string is not used in this PR? or maybe create another PR for using f-string? What do you think?